types/ramda/index.d.ts

```diff
@@ -540,245 +539,36 @@
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => T
-    ] | [
-        (x0: V0) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => T
-    ];
-
-    type ComposeWithFns<V0, T> = [
-        (x0: V0) => T
-    ] | [
-        (x: any) => T,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ] | [
-        (x: any) => T,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: any) => any,
-        (x: V0) => any
-    ];
-
-    interface AssocPartialOne<K extends keyof any> {
-        <T>(val: T): <U>(obj: U) => Record<K, T> & U;
-        <T, U>(val: T, obj: U): Record<K, T> & U;
-    }
+    type Arity0Fn                                                                   = Tools.Arity0Fn;
+    type Arity1Fn                                                                   = Tools.Arity1Fn;
+    type Arity2Fn                                                                   = Tools.Arity2Fn;
+    type AssocPartialOne<K extends keyof any>                                       = Tools.AssocPartialOne<K>;
+    type ComposeWithFns<V0, R>                                                      = Tools.ComposeWithFns<V0, R>;
+    type Dictionary<A>                                                              = Tools.Dictionary<A>;
+    type Evolvable<E extends Evolver>                                               = Tools.Evolvable<E>;
+    type Evolve<O extends Evolvable<E>,  E extends Evolver>                         = Tools.Evolve<O, E>;
```

Comment:
I believe you can use a TS namespace-import for most of these. That syntax means you could avoid retyping the various type parameters and constraints. The only time you wouldn't be able to use it are cases where there's a *value* with the same name.

```suggestion
    export import Evolve = Tools.Evolve;
```

This kind of change isn't necessary for me to approve this, so if you'd rather leave this as-is let me know and I can merge.

Reply:
I forgot to mention that I tried this in the beginning and that `declare let R: R.Static;` is blocking here. I believe this is kind of a trick to enable importing of a few utilities at a time (eg. `import {map} from "ramda"`) while keeping an interface (for convenience, I guess).